docs/integrations/jupiter.md

```diff
@@ -0,0 +1,438 @@
+---
+title: Jupiter
+head:
+  - - meta
+    - name: title
+      content: Solana Cookbook | Swap tokens using Jupiter
+  - - meta
+    - name: og:title
+      content: Solana Cookbook | Swap tokens using Jupiter
+  - - meta
+    - name: description
+      content: Jupiter is the key liquidity aggregator for Solana, offering the widest range of tokens and best route discovery between any token pair.
+  - - meta
+    - name: og:description
+      content: Jupiter is the key liquidity aggregator for Solana, offering the widest range of tokens and best route discovery between any token pair.
+  - - meta
+    - name: og:image
+      content: https://solanacookbook.com/cookbook-sharing-card.png
+  - - meta
+    - name: og:image:alt
+      content: Solana splash card
+  - - meta
+    - name: twitter:card
+      content: summary
+  - - meta
+    - name: twitter:site
+      content: "@solanacookbook"
+  - - meta
+    - name: twitter:image
+      content: "https://solanacookbook.com/cookbook-sharing-card.png"
+  - - meta
+    - name: robots
+      content: index,follow,noodp
+  - - meta
+    - name: googlebot
+      content: index,follow
+---
+
+# Jupiter
+
+Jupiter is the key liquidity aggregator for Solana, offering the widest range of tokens and best route discovery between any token pair.
+
+## How to use Jupiter Core Library
+
+Jupiter Core SDK is used to interact with the Jupiter on-chain programs, which gives us a list of best possible routes for performing a token swap.
+
+### Installation
+
+@jup-ag/core is the Core package used to interact with jupiter on-chain programs to perform swaps between two possible token pairs.
+
+<CodeGroup>
+  <CodeGroupItem title="YARN" active>
+
+```bash
+yarn add @jup-ag/core
+```
+
+  </CodeGroupItem>
+
+  <CodeGroupItem title="NPM">
+
+```bash
+npm install @jup-ag/core
+```
+
+  </CodeGroupItem>
+</CodeGroup>
+
+### Fetching Token list from Jupiter
+
+All the possible tokens that can be swapped with jupiter for a given network is being fetched.
+
+<SolanaCodeGroup>
+  <SolanaCodeGroupItem title="TS" active>
+
+  <template v-slot:default>
+
+@[code](@/code/jupiter/token-list/main.en.ts)
+
+  </template>
+
+  <template v-slot:preview>
+
+@[code](@/code/jupiter/token-list/main.preview.en.ts)
+
+  </template>
+
+  </SolanaCodeGroupItem>
+
+</SolanaCodeGroup>
+
+### Loading the Jupiter instance
+
+Jupiter instace is being created with the provided configurations. There are alot of options present to tinker with to know more about it. go [here](https://docs.jup.ag/jupiter-core/full-guide)
```

Comment:
What is a jupiter instance?

Reply:
Jupiter instance are instantiated with some essentials parameters, Connection, cluster, and User's wallet.

1. We want to allow user to specify their own RPC, be it public or private RPC
2. To support different env
3. To generate/sign the required txs


Once instantiated, user can access the instance's underlying function which powers the swap.
1. Getting routeMap `jupiter.getRouteMap()`, to know which pairs are swappable on Jupiter
2. Computing route `jupiter.computeRoute()`, to get a list of routes/path that can be used
3. To Swap `jupiter.execute()`, performing swap